### PR TITLE
[Feature, NOMERGE] RLHF Rollouts (reopened)

### PR DESCRIPTION
**Pull request by vmoens**
See https://github.com/pytorch/rl/pull/1315 for discussion

### CODE REVIEW

**Comment by vmoens** [comment]
@tcbegley I think we should merge/rebase main into this

**Comment by tcbegley** [comment]
I've rebased and fixed up the tests. There's still the changes from #1328 visible in the history here because I need the reward model for the rollout. Once that's merged (I think it's ready), I'll do a final merge / rebase and we should be ready here too.

File: torchrl/data/rlhf/utils.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+                sample_log_prob: Tensor(shape=torch.Size([4, 50, 1]), device=cpu, dtype=torch.float32, is_shared=False)},
+            batch_size=torch.Size([4, 50]),
+            device=cpu,
+            is_shared=False)
+    """
+
+    EOS_TOKEN_ID = 50256
+
+    def __init__(
+        self, model, ref_model, reward_model, max_new_tokens=50, score_clip=10.0
```
Comment (vmoens) [inline torchrl/data/rlhf/utils.py:89]: long term we may want to make this reward model optional and leave to the user the option of computing reward at a different time (eg when populating the replay buffer)

**Review by vmoens — COMMENTED** [review]
I can't approve it bc it's my PR but LGTM anyway!

**Review by apbard — APPROVED** [review]
LGTM! thanks

**Pull request merged**